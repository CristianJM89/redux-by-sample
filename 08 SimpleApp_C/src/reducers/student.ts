import {actionsEnums} from '../common/actionsEnums';
import objectAssign = require('object-assign');
import {StudentEntity} from '../model/student';

class StudentState  {
  studentsList : StudentEntity[];
  editingStudent : StudentEntity;

  public constructor()
  {
    this.studentsList = [];
    this.editingStudent = new StudentEntity();
  }
}

export const studentReducer =  (state : StudentState = new StudentState(), action) => {
      switch (action.type) {
        case actionsEnums.STUDENTS_GET_LIST_REQUEST_COMPLETED:
           return handleGetStudentList(state, action.payload);
       case actionsEnums.STUDENT_GET_STUDENT_REQUEST_COMPLETED:
          return handleGetStudent(state, action.payload);

      }

      return state;
};

const handleGetStudentList = (state : StudentState, payload : StudentEntity[]) => {
  const newState = objectAssign({}, state, {studentsList: payload});
  return newState;
}


const handleGetStudent = (state : StudentState, payload : StudentEntity[]) => {
  const newState = objectAssign({}, state, {editingStudent: payload});
  return newState;
}
