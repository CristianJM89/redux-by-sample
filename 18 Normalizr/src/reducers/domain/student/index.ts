import { combineReducers } from 'redux';
import {byId} from './byId';
import {allIds} from './list';
import { StudentNormalized } from '../../../model/normalized/studentNormalized';

export interface StudentDomain {
  byId: {[id: number] : StudentNormalized},
  allIds: number[];
};

export const studentDomain = combineReducers<StudentDomain>({
  byId,
  allIds
});
