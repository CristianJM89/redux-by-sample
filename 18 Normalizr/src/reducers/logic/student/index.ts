import { combineReducers } from 'redux';
import { edit, EditState } from './edit';
import { StudentNormalized } from '../../../model/normalized/studentNormalized';

export interface StudentLogic {
  edit: EditState;
};

export const student = combineReducers<StudentLogic>({
  edit
});
