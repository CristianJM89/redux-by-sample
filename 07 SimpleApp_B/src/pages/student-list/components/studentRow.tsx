import * as React from 'react';
import {StudentEntity} from '../../../model/student';

interface Props {
  student : StudentEntity;
}

export const StudentRowComponent = (props : Props) => {
  return (
    <tr>
      <td>
       {(props.student.gotActiveTraining)
        ?
        <span className="glyphicon glyphicon-ok" aria-hidden="true"></span>
        :
        null}
      </td>
      <td>
        <span>{props.student.fullname}</span>
      </td>
      <td>
        <span>{props.student.email}</span>
      </td>
      <td>
        <span className="glyphicon glyphicon-pencil" aria-hidden="true"></span>
        <span className="glyphicon glyphicon-trash" aria-hidden="true"></span>
      </td>

    </tr>
  );
}
