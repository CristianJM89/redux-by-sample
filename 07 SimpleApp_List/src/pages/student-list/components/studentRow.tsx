import * as React from "react";
import { StudentEntity } from "../../../model/student";

interface Props {
  student: StudentEntity;
}

export const StudentRowComponent = (props: Props) => {
  return (
    <tr>
      <td>
        {
          (props.student.gotActiveTraining)
          ?
          <span className="glyphicon glyphicon-ok" aria-hidden="true" />
          :
          null
        }
      </td>
      <td>
        <span>{props.student.fullname}</span>
      </td>
      <td>
        <span>{props.student.email}</span>
      </td>
      <td>
        <span className="glyphicon glyphicon-pencil" aria-hidden="true" />
        <span className="glyphicon glyphicon-trash" aria-hidden="true" />
      </td>
    </tr>
  );
};
