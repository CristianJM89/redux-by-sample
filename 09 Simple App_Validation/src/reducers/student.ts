import { StudentErrors } from "../model/studentErrors";
import { IStudentFieldValueChangedCompletedPayload } from "../pages/student-detail/actions/studentFieldValueChangedCompleted";

import { actionsEnums } from "../common/actionsEnums";
import objectAssign = require("object-assign");
import { StudentEntity } from "../model/student";

class StudentState  {
  studentsList: StudentEntity[];
  editingStudent: StudentEntity;
  editingStudentErrors: StudentErrors;

  public constructor() {
    this.studentsList = [];
    this.editingStudent = new StudentEntity();
    this.editingStudentErrors = new StudentErrors();
  }
}

export const studentReducer =  (state: StudentState = new StudentState(), action) => {
  switch (action.type) {
    case actionsEnums.STUDENTS_GET_LIST_REQUEST_COMPLETED:
      return handleGetStudentList(state, action.payload);
    case actionsEnums.STUDENT_GET_STUDENT_REQUEST_COMPLETED:
      return handleGetStudent(state, action.payload);
    case actionsEnums.STUDENT_FIELD_VALUE_CHANGED_COMPLETED:
      return handleFieldValueChanged(state, action.payload);
  }
  return state;
};

const handleGetStudentList = (state: StudentState, payload: StudentEntity[]) => {
  const newState = objectAssign({}, state, {studentsList: payload});
  return newState;
};

const handleGetStudent = (state: StudentState, payload: StudentEntity[]) => {
  const newState = objectAssign({}, state, {editingStudent: payload});
  return newState;
};

const handleFieldValueChanged = (state: StudentState, payload: IStudentFieldValueChangedCompletedPayload) => {
  const newStudent = objectAssign({}, state.editingStudent, {[payload.fieldName]: payload.value});
  const newStudentErrors = objectAssign({}, state.editingStudentErrors, {[payload.fieldName]: payload.fieldValidationResult});
  return objectAssign({}, state, {editingStudent: newStudent, editingStudentErrors: newStudentErrors});
};
