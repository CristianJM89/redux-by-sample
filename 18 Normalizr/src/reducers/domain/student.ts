import { combineReducers } from 'redux';
import { actionsEnums } from "../../common/actionsEnums";
import objectAssign = require("object-assign");
import { StudentEntity } from "../../model/student";
import { StudentErrors } from "../../model/studentErrors";


const byId = (state = {}, action) => {
  switch (action.type) {
    case actionsEnums.STUDENTS_GET_LIST_REQUEST_COMPLETED:
      return objectAssign({}, state, action.payload.entities.students);
  }
  return state;
}


const allIds =  (state: any = [], action) => {
  switch (action.type) {
    case actionsEnums.STUDENTS_GET_LIST_REQUEST_COMPLETED:
      return action.payload.result;
  }
  return state;
};

export const studentDomain = combineReducers({
  byId,
  allIds
});

export const getStudent = (state, id) => state[id];
export const getIds = (state) => state;
