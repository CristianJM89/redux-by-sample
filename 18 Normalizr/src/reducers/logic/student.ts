import { actionsEnums } from "../../common/actionsEnums";
import objectAssign = require("object-assign");
import { StudentEntity } from "../../model/student";
import { StudentErrors } from "../../model/studentErrors";
import { IStudentFieldValueChangedCompletedPayload } from "../../pages/student-detail/actions/studentFieldValueChangedCompleted";

class StudentState  {
  editingStudent: StudentEntity;
  editingStudentErrors: StudentErrors;

  public constructor() {
    this.editingStudent = new StudentEntity();
    this.editingStudentErrors = new StudentErrors();
  }
}

export const studentReducer =  (state: StudentState = new StudentState(), action) => {
  switch (action.type) {
    case actionsEnums.STUDENT_GET_STUDENT_REQUEST_COMPLETED:
      return handleGetStudent(state, action.payload);
    case actionsEnums.STUDENT_FIELD_VALUE_CHANGED_COMPLETED:
      return handleFieldValueChanged(state, action.payload);
  }
  return state;
};

const handleGetStudent = (state: StudentState, payload: StudentEntity[]) => {
  const newState = objectAssign({}, state, {editingStudent: payload});
  return newState;
};

const handleFieldValueChanged = (state: StudentState, payload: IStudentFieldValueChangedCompletedPayload) => {
  const newStudent = objectAssign({}, state.editingStudent, {[payload.fieldName]: payload.value});
  const newStudentErrors = objectAssign({}, state.editingStudentErrors, {[payload.fieldName]: payload.fieldValidationResult});
  return objectAssign({}, state, {editingStudent: newStudent, editingStudentErrors: newStudentErrors});
};
