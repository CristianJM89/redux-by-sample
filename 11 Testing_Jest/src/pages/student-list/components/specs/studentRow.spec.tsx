import * as React from 'react';
const renderer: any = require('react-test-renderer');
import {StudentRowComponent} from '../studentRow';
import { StudentEntity } from '../../../../model/student';

describe('StudentRowComponent', () => {
  it('Should render a row with a given name and email', () => {
    //Arrange
    const student = new StudentEntity();
    student.id = 2;
    student.gotActiveTraining = true;
    student.fullname = "John Doe";
    student.email = "john@email.com";

    //Act
    const tree = renderer.create(
      <StudentRowComponent
        student={student}
        editStudent={() => {}}
       />
    ).toJSON();

    //Assert
    expect(tree).toMatchSnapshot();
  });

  it('Should interact to the click on edit student and returns as param 2 student Id', () => {
    //Arrange
    const student = new StudentEntity();
    student.id = 2;
    student.gotActiveTraining = true;
    student.fullname = "John Doe";
    student.email = "john@email.com";

    const onEditStudentMock = jest.fn();

    //Act
    const tree = renderer.create(
      <StudentRowComponent
        student={student}
        editStudent={onEditStudentMock}
       />
    ).toJSON();

    const tdContainingDivClickable = tree.children[3];
    const divClickable = tdContainingDivClickable.children[0];

    divClickable.props.onClick();

    //Assert
    expect(tree).toMatchSnapshot();
    expect(onEditStudentMock).toHaveBeenCalled();
    expect(onEditStudentMock).toHaveBeenCalledWith(student.id);
  });
});
