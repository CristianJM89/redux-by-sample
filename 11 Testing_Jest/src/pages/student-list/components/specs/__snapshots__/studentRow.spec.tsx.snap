exports[`StudentRowComponent Should interact to the click on edit student and returns as param 2 student Id 1`] = `
<tr>
  <td>
    <span
      aria-hidden="true"
      className="glyphicon glyphicon-ok" />
  </td>
  <td>
    <span>
      John Doe
    </span>
  </td>
  <td>
    <span>
      john@email.com
    </span>
  </td>
  <td>
    <div
      onClick={[Function]}>
      <span
        aria-hidden="true"
        className="glyphicon glyphicon-pencil" />
    </div>
    <span
      aria-hidden="true"
      className="glyphicon glyphicon-trash" />
  </td>
</tr>
`;

exports[`StudentRowComponent Should render a row with a given name and email 1`] = `
<tr>
  <td>
    <span
      aria-hidden="true"
      className="glyphicon glyphicon-ok" />
  </td>
  <td>
    <span>
      John Doe
    </span>
  </td>
  <td>
    <span>
      john@email.com
    </span>
  </td>
  <td>
    <div
      onClick={[Function]}>
      <span
        aria-hidden="true"
        className="glyphicon glyphicon-pencil" />
    </div>
    <span
      aria-hidden="true"
      className="glyphicon glyphicon-trash" />
  </td>
</tr>
`;
